ngraph/test/backend/reorg_yolo.in.cpp

```diff
@@ -0,0 +1,138 @@
+        73, 75, 77, 79, 89, 91, 93, 95, 105, 107, 109, 111, 121, 123, 125, 127};
+    // in_shape [N,C,H,W] -> out_shape [N, C*stride*stride, H/stride, W/stride]
+    Shape expected_shape = Shape{
+        in_shape[0], in_shape[1] * stride * stride, in_shape[2] / stride, in_shape[3] / stride};
+
+    auto test_case = test::TestCase<TestEngine>(fun);
+    test_case.add_input<float>(inputs);
+    test_case.add_expected_output<float>(expected_shape, expected_result);
+    test_case.run();
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, reorg_yolo_stride_3)
+{
+    // in_shape [N,C,H,W]
+    const auto in_shape = Shape{1, 9, 3, 3};
+    auto p = make_shared<op::Parameter>(element::f32, in_shape);
+    size_t stride = 3;
+    auto reorg_yolo = make_shared<op::v0::ReorgYolo>(p, Strides{stride});
+    auto fun = make_shared<Function>(OutputVector{reorg_yolo}, ParameterVector{p});
+
+    std::vector<float> inputs(81);
+    std::iota(inputs.begin(), inputs.end(), 0);
+    std::vector<float> expected_result{
+        0,  3,  6,  27, 30, 33, 54, 57, 60, 1,  4,  7,  28, 31, 34, 55, 58, 61, 2,  5,  8,
+        29, 32, 35, 56, 59, 62, 9,  12, 15, 36, 39, 42, 63, 66, 69, 10, 13, 16, 37, 40, 43,
+        64, 67, 70, 11, 14, 17, 38, 41, 44, 65, 68, 71, 18, 21, 24, 45, 48, 51, 72, 75, 78,
+        19, 22, 25, 46, 49, 52, 73, 76, 79, 20, 23, 26, 47, 50, 53, 74, 77, 80};
+    // in_shape [N,C,H,W] -> out_shape [N, C*stride*stride, H/stride, W/stride]
+    Shape expected_shape = Shape{
+        in_shape[0], in_shape[1] * stride * stride, in_shape[2] / stride, in_shape[3] / stride};
+
+    auto test_case = test::TestCase<TestEngine>(fun);
+    test_case.add_input<float>(inputs);
+    test_case.add_expected_output<float>(expected_shape, expected_result);
+    test_case.run();
+}
+
+NGRAPH_TEST(${BACKEND_NAME}, reorg_yolo_catch_small_shape_stride)
+{
+    // Throw error test: For [N, C, H, W] input shape, C >= (stride*stride) is required.
+    // in_shape [N,C,H,W]
+    const auto in_shape = Shape{1, 1, 4, 4};
+    auto p = make_shared<op::Parameter>(element::f32, in_shape);
+    size_t stride = 2;
+    auto reorg_yolo = make_shared<op::v0::ReorgYolo>(p, Strides{stride});
+    auto fun = make_shared<Function>(OutputVector{reorg_yolo}, ParameterVector{p});
+
+    std::vector<float> inputs{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
+    std::vector<float> fake_output(16);
+
+    // in_shape [N,C,H,W] -> out_shape [N, C*stride*stride, H/stride, W/stride]
+    Shape expected_shape = Shape{
+        in_shape[0], in_shape[1] * stride * stride, in_shape[2] / stride, in_shape[3] / stride};
+
+    try
+    {
+        auto test_case = test::TestCase<TestEngine>(fun);
+        test_case.add_input<float>(inputs);
+        test_case.add_expected_output<float>(expected_shape, fake_output);
+        test_case.run();
+
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Incompatible stride was not detected.";
```

Comment:
Did you consider using [exception assertions](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#exception-assertions) ?

Reply:
I think `ASSERT_THROW(test_case.run(), ngraph_error)` is too general `ASSERT_ANY_THROW()` even more.
The test will pass even the error is not about stride,